src/GraphQLParser/Visitors/ASTVisitor.cs

```diff
@@ -598,6 +598,8 @@ protected virtual async ValueTask VisitInputObjectTypeExtensionAsync(GraphQLInpu
     /// <param name="context">Context passed into all INodeVisitor.VisitXXX methods.</param>
     public virtual ValueTask VisitAsync(ASTNode? node, TContext context)
     {
+        context.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
One-line fix :)

Reply:
Is this ever called recursively?

Reply:
> Is this ever called recursively?

Of course.

Reply:
Then I’d probably not add it.

Reply:
Whatever you think. I’ve never timed how slow that method is.

Reply:
?

Reply:
That method dispatches call to some other method and so on.